### DISCUSSION TITLE
Help with ST7789 2.8" touch 240x320 TFT with ESP Wroom 32

### DISCUSSION THREAD

**Discussion by thegiantpanda001** [Bodmer/TFT_eSPI#2177] [Q&A - General]
Hi, ESP32 n00b here, I have been trying to interface this lcd I got from aliexpress with an esp I got from a friend.

The esp I have is a no-brand Wroom 32, 38 pin, configured as NodeMCU-32 in Arduino IDE (I guessed based on pinout diagrams), works fine with various sensors.
The pin out is of the 38 pin type like below:

![image](https://user-images.githubusercontent.com/118244085/202049803-511fd90a-93b7-4200-9664-11a3b51e8897.png)

The LCD page is: 
https://www.aliexpress.com/item/4001282467099.html 
![image](https://user-images.githubusercontent.com/118244085/202047286-6239e9db-fdb0-4dd0-a3af-259108a5e7f6.png)
but no specs are given, after contacting supplier they said it was an ST7789 and provided a sample program but for a different MCU
[2.4-2.8_ST7789spiORIfastRGB(QQWW).c.txt](https://github.com/Bodmer/TFT_eSPI/files/10016967/2.4-2.8_ST7789spiORIfastRGB.QQWW.c.txt)

I have tried to use the TFT_eSPI and also the Adafruit TFT libraries, configured for ST7789, with no success.
I have tried both the built-in SPI pins and custom-defined pins. All I am able to do is turn on the backlight.

My User_Setup.h file is here:
[User_Setup.h.txt](https://github.com/Bodmer/TFT_eSPI/files/10016878/User_Setup.h.txt)

I run read_user_setup and get the below:

```
[code]
TFT_eSPI ver = 2.4.79
Processor    = ESP32
Frequency    = 240MHz
Transactions = Yes
Interface    = SPI
Display driver = 7789
Display width  = 240
Display height = 320

MOSI    = GPIO 23
MISO    = GPIO 19
SCK     = GPIO 18
TFT_CS   = GPIO 5
TFT_DC   = GPIO 32
TFT_RST  = GPIO 35

TFT_BL           = GPIO 25
TFT_BACKLIGHT_ON = HIGH

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 27.00
[/code]

```
DC, RST, and BL are my own defined pins, the rest are default hardware pins.
Would appreciate any help with this as I have tried many times without success.

**Comment by sparx266** [comment]
In your setup, you haven't set any pin numbers for the definitions.

Yours looks like this:

#define TFT_MISO MISO
#define TFT_MOSI MOSI
#define TFT_SCLK SCK
#define TFT_CS   SS  // Chip select control pin
#define TFT_DC   32  // Data Command control pin
#define TFT_RST  35  // Reset pin (could connect to RST pin)
// #define TFT_RST  -1  // Set TFT_RST to -1 if display RESET is connected to ESP32 board RST

Try this:

#define TFT_MISO 19
#define TFT_MOSI 23
#define TFT_SCLK 18
#define TFT_CS   5  // Chip select control pin
#define TFT_DC   32  // Data Command control pin
#define TFT_RST   4  // Reset pin (could connect to RST pin)
#define TFT_RST  35  // Set TFT_RST to -1 if display RESET is connected to ESP32 board RST

I would guess you have an ILI9341 display with an XP2046 touch driver, but that is just a guess.

Try the above and report back....

You can get there! 

:-)

**Reply by thegiantpanda001** [reply]
Hi sparx266, thanks for your reply.

The MISO, MOSI, SCK and SS pins are already defined internally for ESP32 Default SPI Pins, so these should be ok (otherwise sketch would not compile).
[https://randomnerdtutorials.com/esp32-spi-communication-arduino/](url)
I have already tried using numbers directly as well, and it doesn't make a difference.
Tbh the supplier said ST7789 but I'm still not sure exactly what model the LCD is, could also be a hardware issue.
I will go and try other drivers just to be sure.

It would be good if I can send a command to the LCD and read some data back to confirm the SPI connection is working.
Not quite sure how to do this in the code and with what pin but I heard it is possible to read the LCD registers, also from ST7789 datasheet there is a read ID command or something (0x04h...) any help there would be good!

Or I might just buy another IL9341 LCD...

Thanks!

**Reply by thegiantpanda001** [reply]
Finally, I got it working! The TFT is an ST7789V as the supplier suggested. The supplier provided some basic sample code that writes some commands to the LCD without using any libraries. The problem was that on my dodgy esp32 the G33 pin is erroneously labeled G23. G23 is supposed to be the default MOSI pin. Once I connected to the correct G23 MOSI pin, the sample code worked and showed a test screen. I am now trying the example sketches from the library, which can show a colored screen. Will test some more.

**Reply by pushkarmondhe** [reply]
hello @thegiantpanda001  i am working on st7789v display i am connecting to esp32 wroom but not display anything please help me reslove this issue.i am connceted mosi 32,sck 18,data command pin 15,backlight 4,reset 5
